donkeycar/parts/oak_d.py

```diff
@@ -0,0 +1,329 @@
+                    (WIDTH, HEIGHT), (self.width, self.height))
+            )
+
+        self.pipeline = None
+        if self.enable_depth or self.enable_rgb:
+            self.pipeline = depthai.Pipeline()
+
+            device_info = self.get_depthai_device_info(device_id)
+
+            if self.enable_depth:
+                self.setup_depth_camera(WIDTH, HEIGHT)
+
+            if self.enable_rgb:
+                self.setup_rgb_camera(WIDTH, HEIGHT)
+
+            self.oak_d_device = depthai.Device(self.pipeline, device_info)
+
+        # initialize frame state
+        self.color_image = None
+        self.depth_image = None
+        self.frame_count = 0
+        self.start_time = time.time()
+        self.frame_time = self.start_time
+
+        self.running = True
+
+    # Taken from the demo application.
+    def get_depthai_device_info(self, device_id: string):
+        device_infos = depthai.Device.getAllAvailableDevices()
+        if len(device_infos) == 0:
+            raise RuntimeError("No DepthAI (Oak-D-Lite) device (camera) found!")
+        else:
+            print("Available devices:")
+            for i, deviceInfo in enumerate(device_infos):
+                print(f"[{i}] {deviceInfo.getMxId()} [{deviceInfo.state.name}]")
+
+            # Set the deviceId to "list" in order to list the connected devices' ids.
+            if device_id == "list":
+                raise SystemExit(0)
+            elif device_id is not None:
+                matching_device = next(filter(lambda info: info.getMxId() == device_id, device_infos), None)
+                if matching_device is None:
+                    raise RuntimeError(f"No DepthAI device found with id matching {device_id} !")
+                return matching_device
+            elif len(device_infos) == 1:
+                return device_infos[0]
+            else:
+                val = input("Which DepthAI Device you want to use: ")
+                try:
+                    return device_infos[int(val)]
+                except:
+                    raise ValueError("Incorrect value supplied: {}".format(val))
+
+    def setup_depth_camera(self, width, height):
+        # Set up left and right cameras
+        mono_left = self.get_mono_camera(self.pipeline, True)
+        mono_right = self.get_mono_camera(self.pipeline, False)
+
+        # Combine left and right cameras to form a stereo pair
+        stereo: depthai.node.StereoDepth = self.get_stereo_pair(self.pipeline, mono_left, mono_right)
+
+        # Define and name output depth map
+        xout_depth = self.pipeline.createXLinkOut()
+        xout_depth.setStreamName("depth")
+
+        stereo.depth.link(xout_depth.input)
+
+    def setup_rgb_camera(self, width, height):
+        cam_rgb = self.pipeline.create(depthai.node.ColorCamera)
+
+        res = depthai.ColorCameraProperties.SensorResolution.THE_1080_P
+
+        cam_rgb.setResolution(res)
+        # cam_rgb.setInterleaved(False)
```

Comment:
Please remove commented code. You can add suggestions about the code in the doc string of the function.